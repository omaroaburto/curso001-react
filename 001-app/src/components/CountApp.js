import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CountApp = ({value}) => {
    //handAdd
    const [count, setCount] = useState(value);
    const handleAdd = () =>{ 
        //forma 1
        setCount(count+1);
        //forma 2
        //setCount((c)=> c+1);
    } 
    const handleSubtract = () =>{ 
        //forma 1
        if(count!==0)
            setCount(count-1); 
    } 
    const handleReset = () =>{ 
        //forma 1
        setCount(value); 
    } 
    return (
        <>
            <h1>CounterApp</h1>
            <span>{count}</span> 
            <br/>
            <button onClick={handleAdd}>+1</button> 
            <button onClick={handleSubtract}>-1</button> 
            <button onClick={handleReset}>Reset</button> 
        </>
    );
}

CountApp.propTypes = {
    value: PropTypes.number
}
CountApp.defaultProps = {
    value: 10
}
export default CountApp
