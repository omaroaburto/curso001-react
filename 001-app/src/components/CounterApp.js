import React from 'react'
import PropTypes from 'prop-types';
//ejemplo de pros
const CounterApp = ({numero}) => {
    return (
        <>
            <h1>Hola mundo</h1>
            <span>{numero}</span>
        </>
    )
}

CounterApp.propTypes={
    numero : PropTypes.number.isRequired
};

export default CounterApp;