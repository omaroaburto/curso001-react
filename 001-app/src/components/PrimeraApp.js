import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, mensaje}) => {
    return (
        <>
            <h1>{saludo}</h1>
            <span>{mensaje}</span>   
        </>
    )
}

PrimeraApp.propTypes={
    saludo : PropTypes.string.isRequired,
    mensaje: PropTypes.string.isRequired
};

PrimeraApp.defaultProps ={
    saludo: 'Hola gente',
    mensaje: 'Mi nombre es omaro'
}

export default PrimeraApp;