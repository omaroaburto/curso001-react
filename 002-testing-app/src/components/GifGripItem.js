import React from 'react'
import PropTypes from 'prop-types';
export const GifGripItem = ({id, title, url}) => {
    return (
        <div key={id} className="item-caja">
            <img 
                key={id} 
                src={url} 
                alt={title}
            />   
        </div>  
    )
}

GifGripItem.propTypes ={
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired 
}

