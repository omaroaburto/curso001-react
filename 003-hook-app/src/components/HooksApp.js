import React from 'react'

export const HooksApp = () => {
    return (
        <div>
            <h1 className="mt-4">Hola mundo</h1>
        </div>
    )
}
