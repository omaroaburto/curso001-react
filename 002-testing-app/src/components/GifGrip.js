import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif'; 
import { GifGripItem } from './GifGripItem';
import PropTypes from 'prop-types';
export const GifGrip =  ({category}) => { 
    const {data:imagenes, loading} = useFetchGif(category); 

    return (
        <>
            <h3>{category.toUpperCase()}</h3> 
            {loading && <span>Loading...</span>}
            <div className="caja"> 
                {
                    imagenes.map(({url, id, title}) =>{
                            return (
                                <GifGripItem key={Math.random()} url={url} id={id} title={title}/>  
                            );
                    })
                }
            </div> 
        </>
    )
}

GifGrip.propTypes = {
    category: PropTypes.string.isRequired
}
