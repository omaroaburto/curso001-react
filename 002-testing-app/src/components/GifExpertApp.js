import React, {useState} from 'react'
import AddCategory from './AddCategory';
import { GifGrip } from './GifGrip';

export const GifExpertApp = ({elements=['gatos', 'perros']}) => {
    const [categories, setCategories] = useState(elements);

    return (
        <>
          <h1>GifExpertApp</h1>
          <AddCategory setCategories = {setCategories} categories = {categories}/>
          <hr/> 
          <ul>
            {categories.map(category=>{
                return (<GifGrip 
                    category={category}
                    key={category}
                />);
            })}
          </ul>  
        </>
    )
}

 
