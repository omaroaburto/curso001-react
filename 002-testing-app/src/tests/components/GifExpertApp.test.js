import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../../components/GifExpertApp';
describe('Pruebas del archivo GifExpertApp.test.js',()=>{
    test('Pruebas del componente <GifExpertApp/>',()=>{
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe mostrar una lista de categorias',()=>{
        const categories = ['Rock', 'Queen', 'Animales'];
        const wrapper = shallow(<GifExpertApp elements={categories}/>);
        expect(wrapper.find('GifGrip').length).toBe(categories.length)
    });
});