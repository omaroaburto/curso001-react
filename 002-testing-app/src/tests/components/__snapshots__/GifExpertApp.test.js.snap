// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo GifExpertApp.test.js Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas del archivo GifExpertApp.test.js Pruebas del componente <GifExpertApp/> 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    categories={
      Array [
        "gatos",
        "perros",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrip
      category="gatos"
      key="gatos"
    />
    <GifGrip
      category="perros"
      key="perros"
    />
  </ul>
</Fragment>
`;
